Dockerfile

```diff
@@ -0,0 +1,111 @@
+RUN npm install es6-promise
+RUN npm install --unsafe-perm
+
+RUN node_modules/webpack/bin/webpack.js -p
+
+FROM centos:7
+
+RUN rpm -iUvh http://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/e/epel-release-7-13.noarch.rpm
+
+RUN yum -y update
+RUN yum -y groupinstall "Development tools"
+RUN yum -y install wget gcc-c++ openssl-devel \
+                   postgresql-devel mysql-devel \
+                   git libffi-devel libxml2-devel libxml2 \
+                   libxslt-devel zlib1g-dev libxslt http-parser uwsgi
+
+# Install Python...
+
+ENV PYTHON_VER=3.6
+ENV PYTHON_PRG=/usr/bin/python${PYTHON_VER}
+ENV PYTHON_ENV=/opt/.venv
+ENV PYTHON_LIB=${VIRTUAL_ENV}/lib/python${PYTHON_VER}
+
+RUN echo -e \
+    "\tPYTHON VERSION : $PYTHON_VER\n" \
+    "\tPYTHON PROGRAM : $PYTHON_PRG\n" \
+    "\tPYTHON VIRTUAL : $PYTHON_ENV\n" \
+    "\tPYTHON LIBRARY : $PYTHON_LIB\n"
+
+RUN yum -y install python${PYTHON_VER//.} python${PYTHON_VER//.}-pip python${PYTHON_VER//.}-devel python${PYTHON_VER//.}-virtualenv
+RUN yum -y install uwsgi-plugin-python${PYTHON_VER//.}
+
+RUN ${PYTHON_PRG} -m virtualenv --python=${PYTHON_PRG} ${PYTHON_ENV}
+ENV PATH="$PYTHON_ENV/bin:$PATH"
+
+# install locale
+RUN localedef -c -f UTF-8 -i en_US en_US.UTF-8
+ENV LC_ALL=en_US.utf-8
+ENV LANG=en_US.utf-8
+
+# Prepare application backend...
+
+ENV B2SHARE_INSTANCE_PATH=/opt/var
+
+WORKDIR /opt/app
+
+ADD MANIFEST.in .
+ADD entry_points.txt .
+ADD requirements.txt .
+ADD README.rst .
+ADD setup.py .
+ADD setup.cfg .
+ADD b2share b2share
+
+RUN pip install -r requirements.txt
+
+# Prepare application frontend...
+
+COPY webui/app b2share/modules/b2share_main/static
+
+RUN mv b2share/modules/b2share_main/static/index.html b2share/modules/b2share_main/templates/b2share_main/page.html
+
+# Add generated files into lib sub directories...
+COPY --from=build-deps /opt/app/b2share-bundle.* b2share/modules/b2share_main/static/
+COPY --from=build-deps /opt/node_modules/bootstrap/dist/css/bootstrap.min.* b2share/modules/b2share_main/static/lib/css/
+COPY --from=build-deps /opt/node_modules/bootstrap-grid/dist/grid.min.css b2share/modules/b2share_main/static/lib/css/bootstrap-grid.min.css
+COPY --from=build-deps /opt/node_modules/react-widgets/dist/css/* b2share/modules/b2share_main/static/lib/css/
+COPY --from=build-deps /opt/node_modules/font-awesome/css/* b2share/modules/b2share_main/static/lib/css/
+COPY --from=build-deps /opt/node_modules/react-toggle/style.css b2share/modules/b2share_main/static/lib/css/toggle-style.css
+COPY --from=build-deps /opt/node_modules/bootstrap/dist/fonts/* b2share/modules/b2share_main/static/lib/fonts/
+COPY --from=build-deps /opt/node_modules/react-widgets/dist/fonts/* b2share/modules/b2share_main/static/lib/fonts/
+COPY --from=build-deps /opt/node_modules/font-awesome/* b2share/modules/b2share_main/static/lib/fonts/
+COPY --from=build-deps /opt/node_modules/react-widgets/dist/img/* b2share/modules/b2share_main/static/lib/img/
```

Comment:
Note the use of `b2share/modules/b2share_main/static` which essentially is default value for B2SHARE_STATIC_FOLDER used in factory.py.
Note that B2SHARE V2 uses B2SHARE_WEBUI_PATH for this.

Reply:
If  / when we separate webui from backend codebase we need to adjust these as needed.